zebra/zebra_dplane.c

```diff
@@ -887,6 +896,10 @@ static void dplane_ctx_free_internal(struct zebra_dplane_ctx *ctx)
 	case DPLANE_OP_STARTUP_STAGE:
 	case DPLANE_OP_SRV6_ENCAP_SRCADDR_SET:
 		break;
+	case DPLANE_OP_VLAN_INSTALL:
+		if (ctx->u.vlan_info.vlan_array)
+			XFREE(MTYPE_TMP, ctx->u.vlan_info.vlan_array);
```

Comment:
we've been trying to get away from using MTYPE_TMP like this, as a catch-all. could you find a better mem type?

Reply:
Ack, Quick ques on this, 
since the alloc happens in if_netlink.c and free in zebra_dplane.c, which of the below do you prefer.. 

1) in lib/memory.c/h I define a mtype called VLAN_CHANGE (similar to TMP) and use it in both the files (or)
2) define mtype called VLAN_CHANGE  in zebra_dplane.c and have a function to realloc vlan_array memory called from if_netlink.c

Reply:
I think a model to follow might be MTYPE_RE in zebra: it's DECLAREd in `rib.h`, and used in `zebra_rib.c` and in `zebra_rnh.c` ?

Reply:
Done